posix/include/rtos/alloc.h

```diff
@@ -0,0 +1,172 @@
+/* SPDX-License-Identifier: BSD-3-Clause
```

Comment:
not able to include those headers? duplicate is not a good option.

Reply:
@btian1 this is for host build for testbench/tplg parser/plugin etc where zephyr/xtos doesnt feature

Reply:
so you mean plugin build does not have xtos/zephyr included?

Reply:
we do already use SOF headers for library (testbench, cmocka) builds, cannot we also use them for this? Really should avoid duplicating - a sure path to breakage and confusion...

Reply:
But which set? I think this approach has merits as well. I think we have three options:
1) XTOS build with CONFIG_LIBRARY (use sof/xtos/rtos)
2) Zephyr build with posix as arch (use sof/zephyr/rtos)
3) Native library build (use sof/posix/rtos)

I think (3) involves code duplication, but for a pure library build (that can be linked to any Linux application), I think this approach is cleanest. Especially if we longterm remove the the XTOS build. Using Zephyr posix build is one option, but I don't think that is really ideal for the library use (like an ALSA plugin, or exposing audio modules via some other framerwork like LV2).

Reply:
I would move list.h and other generic (non RTOS api) level headers to a sof/lib (but this could be another patch).

Reply:
@lgirdwood Right, but we don't need XTOS arch code if plugin uses its own rtos implementation.  sof/posix/rtos should be independent of sof/src/arch . cmocka uses it, but it was done before we had a rtos abstraction.

Reply:
not hugely important, but calling these "posix headers" is a bit confusing - don't think all these headers are POSIX